nsis/SectionVersionCheck.nsh

```diff
@@ -0,0 +1,133 @@
+; Copyright (C) 2017 Christian Kenny
+
+Function UninstallPrevious
+
+; Here's how StrContains works:
+; $result_var: This will store our result if the "needle" is found. Otherwise it will return null ("")
+; $needle: String to search for
+; $Haystack: String to look in
+
+; Search for 1.0.0
+ReadRegStr $R7 HKLM "${INSTDIR_REG_KEY}" "Uninst-pcsx2Directory"
+${StrContains} "$4" "1.0.0" "$R7"
+StrCmp $4 "" +1
+${If} $R7 == ""
+${EndIf}
+
+; This will become the primary version check once more stable builds
+ReadRegStr $R1 HKLM "${INSTDIR_REG_KEY}" "DisplayVersion"
+${If} $R1 == ""
+${EndIf}
+
+; Search for 1.4.0
+ReadRegStr $R2 HKLM "${INSTDIR_REG_KEY}" "Uninst-pcsx2 1.4.0Directory"
+${StrContains} "$2" "1.4.0" "$R2"
+    StrCmp $2 "" +1
+${If} $R2 == ""
+${EndIf}
+
+; Search for 1.2.1
+ReadRegStr $R3 HKLM "${INSTDIR_REG_KEY}-r5875" "Uninst-pcsx2-r5875Directory"
+${StrContains} "$3" "1.2.1" "$R3"
+    StrCmp $3 "" +1
+${If} $R3 == ""
+${EndIf}
+
+; Search for 0.9.8
+ReadRegStr $R5 HKLM "${INSTDIR_REG_KEY}-r4600" "Uninst-pcsx2-r4600Directory"
+${StrContains} "$5" "0.9.8" "$R5"
+    StrCmp $5 "" +1
+${If} $R5 == ""
+${EndIf}
+
+; If all cases return null, our work here is done.
+;${If} $R1 == ""
+${If} $R2 == ""
+${AndIf} $R3 == ""
+${AndIf} $R5 == ""
+${AndIf} $R7 == ""
+Goto Done
+${EndIf}
+
+; Installing another version
+    MessageBox MB_ICONEXCLAMATION|MB_OKCANCEL "Another version of PCSX2 is already installed. The current configuration folder in Documents will be duplicated and renamed as PCSX2_backup. Click OK to uninstall PCSX2 or Cancel to abort the setup." IDOK true IDCANCEL false
+true:
+Goto SetUninstPath
+    false:
+        Quit
+
+SetUninstPath:
+${If} $R7 != ""
+    ${AndIf} $R1 != ""
+    ${OrIf} $4 == "1.0.0"
+    Goto ExecNormal
+${Else}
+        Goto +1
+${EndIf}
+
+${If} $R2 != ""
+    ${AndIf} $2 == "1.4.0"
+    Goto Exec1.4.0
+${Else}
+        Goto +1
+${EndIf}
+
+${If} $R3 != ""
+    ${AndIf} $3 == "1.2.1"
+    Goto Exec1.2.1
+${Else}
+        Goto +1
+${EndIf}
+
+${If} $R5 != ""
+    ${AndIf} $5 == "0.9.8"
+    Goto Exec0.9.8
+${Else}
+        Goto +1
+${EndIf}
+
+ExecNormal:
+SetOutPath "$DOCUMENTS"
+CopyFiles /SILENT "$DOCUMENTS\PCSX2" "$DOCUMENTS\PCSX2_backup"
+RMDir /r "$DOCUMENTS\PCSX2"
+    SetOutPath "$TEMP"
```

Comment:
These 4 lines are common in each uninstall case. It'd be nice to avoid duplication.